### ISSUE TITLE
StyleURL does not change after initial load

### ISSUE THREAD

**Issue by justin-mylermedia** [mapbox/mapbox-gl-native-ios#294]
We are running into an issue with the styleURL, we are not able to find a way to change the styleURL after the initial load (besides making the MapView a binding which makes it reload on every change). We have tried to make it a UIViewControllerRepresentable but that did not seem to solve the problem. We are using SwiftUI

### Steps to reproduce
 1. Create a state and binding for the styleURL. This is the code we have used for this.

```swift
@Binding var styleUrl: URL

    func updateUIView(_ uiView: MGLMapView, context: UIViewRepresentableContext<MapView>) {
        updateAnnotations()
        mapView.styleURL = styleUrl
    }
```
 2.  Create a button and change the styleURL on a button click.

### Expected behavior

The styleURL should change when updateUIView gets changed.

### Actual behavior

The modifications to the StyleURL after the map has loaded do not show up. The initial state does get loaded. Whatever we set the styleURL to at the start is what shows up on the map. The updateUIView method does get called and the styleURL seems to get updated. It just does not actually change on the map.

### Configuration

**Mapbox SDK versions:** 5.8.0 - 5.9.0-alpha1
**iOS/macOS versions:** 13.1.1
**Device/simulator models:** iPad 2019
**Xcode version:** 11.4.1 (SwiftUI)

**Comment by 1ec5** [comment]
Can you share more of the code you’re using to set up the map view? I’m curious whether you’re seeing the same map view not updating or whether tapping the button happens to create a different map view that’s initialized with the original style.

**Comment by justin-mylermedia** [comment]
Hey 1ec5, we found the solution to this problem. The issue can be closed.

For anyone else with the same problem it had to do with calling the MGLMapView property we stored at the top of the class. After changing the code above to using the MGLMapView we got from the updateUIView it started to work. We changed the code above to this and now it works:
```swift
@Binding var styleUrl: URL

    func updateUIView(_ uiView: MGLMapView, context: UIViewRepresentableContext<MapView>) {
        updateAnnotations(mapView: uiView)
        uiView.styleURL = styleUrl
    }
```
